skymarshal/skycmd/oidc_flags.go

```diff
@@ -28,6 +28,7 @@ type OIDCFlags struct {
 	HostedDomains      []string    `long:"hosted-domains" description:"List of whitelisted domains when using Google, only users from a listed domain will be allowed to log in"`
 	CACerts            []flag.File `long:"ca-cert" description:"CA Certificate"`
 	InsecureSkipVerify bool        `long:"skip-ssl-validation" description:"Skip SSL validation"`
+  InsecureEnableGroups bool      `long:"enable-groups" description:"Enable OIDC groups"`
```

Comment:
@xtremerui do we want this as an opt-in feature, or should it be opt-out instead? Making it opt-in means it'll be breaking (compared to 6.7.2 - 6.7.3 was breaking in an unintended way)

Reply:
Or do we even need to provide the ability to opt-out?

Reply:
Alternatively, it can be true on default, but also overridable ...

Reply:
Just thought, it would make sense, since upstream dex does this ...

Reply:
Good call. Making it opt-out is better for backward compatibility but at the same time opt-in flag makes user to aware that enabling groups is insecure. 

Given existing concourse users of OIDC connectors still have a chance to add the flag when they upgrading Concourse, IMO keep it opt-in is more aligned to upstream DEX.

Reply:
I don't think enabling groups is actually insecure here. See https://github.com/dexidp/dex/commit/21ab30d20767283b4696c992644432a38fc23849

> There's been some discussion in dexidp/dex#1065 regarding what to do about refreshing groups. As it stands today dex doesn't update any of the claims on refresh (groups would just be another one). The main concern with enabling it is that group claims may change more frequently.

So the insecure behaviour is around using the OIDC connector with refresh tokens, since those are long-lived and groups can change, but that change won't be propagated. But Concourse doesn't support refresh tokens, so it's no different from any other connector (when your groups change, you have to log out and log back in).

Also, being more aligned with upstream Dex doesn't necessarily make sense here - using groups is pretty central to Concourse's auth system, whereas Dex is just meant to be a general purpose identity provider

Reply:
I think, both opt-in and opt-out would make sense. Probably, a question of taste ...

Reply:
@konstl000 in your use case, do you specified `--oidc-scopes` with `groups` ?

Maybe, we could set `InsecureEnableGroups` to true if `--oidc-scopes` contains `groups` since concourse needs both to work with groups claim.

Reply:
@xtremerui Actually not, because we just include groups into the openID scope on the Keycloak side. Probably, there are more people around who do that, considering that other tools (such as HashiCorp Vault) can rely on groups too ...

Reply:
Once I saw that it did not work, I tried to include groups via --oidc-scopes though :-)

Reply:
Makes sense to me - wasn't aware of `--oidc-scopes`. I just don't think it makes much sense to introduce a breaking change that forces operators to opt-in to a feature that, while potentially insecure in general, doesn't affect auth in Concourse

Reply:
I am quite fine with the opt-out option too. Should I change the PR accordingly?

Reply:
> Maybe, we could set `InsecureEnableGroups` to true if `--oidc-scopes` contains `groups` since concourse needs both to work with groups claim. WDYT @aoldershaw

That would be a breaking change too, at least for folks that are as lazy as me and just used to include groups in the openID scope.

Reply:
@konstl000 ok thx!

Reply:
Now it is opt-out and is called --oidc-disable-groups

Reply:
> That would be a breaking change too, at least for folks that are as lazy as me and just used to include groups in the openID scope

Ah sorry, I didn't see https://github.com/concourse/concourse/pull/6436#discussion_r559676991 before I added my most recent comment - so just to be clear, in 6.7.2 and earlier, you didn't need to set `--oidc-scopes` to include `groups` - it picked them up by default?

Reply:
Yes, if the OIDC provider had `groups` in the openID scope, the earlier versions of Concourse would pick them up and use them without the `- -oidc-scopes` flag. This is also how our production Concourse is configured at the moment.

Reply:
> Yes, if the OIDC provider had `groups` in the `openid` scope, the earlier versions of Concourse would pick them up and use them without the `--oidc-scopes` flag. 

So it depends on how OIDC provider returns claims based on scopes. In @konstl000 case Keycloak is configured to return groups claims by one of the default scopes [openid, profile, email] so `--oidc-scopes` is not needed. I agree that `--oidc-scopes` and disable groups claim should then be decoupled to achieve maximum flexibility.